### PR TITLE
BL-12241 Return consistent objects

### PR DESCRIPTION
**Pull request by JoshuaLicense**
| Q                | A
| ---------------- | ---
| Type             | Feature
| Ticket           | BL-12241

**Description of the change:**

- Implement the `AccessTokenInterface` with OIDC elements.
- Return a consistent object from `authenticate` call.
- Format attribute values per the AWS requirements (enables us to use bools as values).
- Add ability to specify token "leeway" to account for clock skew.

### CODE REVIEW

File: src/Client.php [hunk 1]
```diff
@@ -4,15 +4,25 @@
...
 
 class Client implements OAuthClientInterface
 {
+    /**
+     * When checking nbf, iat or expiration times on tokens, we want to provide
+     * some extra leeway time to account for clock skew.
+     *
+     * @var int
+     */
+    public static $leeway = 0;
```
Comment (Hodgy) [inline src/Client.php:24]: Do we need this if it's going to be 0? Or is the plan to figure out what it should be based on testing?

Comment (JoshuaLicense) [reply src/Client.php:24]: Good question, _I think_ we need this to allow consumers to set their own desired leeway. In my opinion, there isn't a "correct" value for this as every scenario will be different depending on infrastructure and risk appetite.

Consumers can (recommended) to set this value while setting up the class in their app container (like [this](https://github.com/dvsa/authentication-examples/pull/10/files#diff-01c87be29887e0b4c02e0fd783f4c529c220d0aa34308ea01458aceaedabc526R206)). Using the skeleton app as an example of clock skew, as we are usually serving the app locally, the time on our local machine and AWS _could_ differ on some occasions (my local environment was 2 seconds behind AWS yesterday).

Comment (Hodgy) [reply src/Client.php:24]: Ah yes the consumes would be be wanting to set it. Had overlooked that.

File: composer.json [hunk 2]
```diff
@@ -14,8 +14,9 @@
     "php": "^7.2.5 || ^8.0",
     "ext-json": "*",
     "aws/aws-sdk-php": "~3.183.7",
-    "dvsa/contracts": "dev-main",
-    "firebase/php-jwt": "^5.2"
+    "dvsa/contracts": "dev-feature-BL-12241-return-consistent-objects",
```
Comment (sdh100shaun) [inline composer.json:17]: Can we not push contracts now?

Comment (JoshuaLicense) [reply composer.json]: I've pointed it back to `dev-main`

**Review by Hodgy — APPROVED** [review]

**Pull request merged**